### DISCUSSION TITLE
UI redesign - status bar is too cramped

### DISCUSSION THREAD

**Discussion by H-G-Hristov** [microsoft/vscode-cpptools#9845] [Ideas]
All important extensions take up status bar space and some of the most important information gets lost.

Basically I use the following extensions in a typical C++ project:

1. git (Lens + Pull requests + GitGraph)
2. CMake Tools
3. C++ Tools

I'd like to be able to see: line, column numbers, file type, clang-tidy/intellisense icon, etc but they get pushed out by the widgets added to the status bar.
It's also important to me to be able to see the git branch I'm on because I mess it up too often.

I remembered that @bobbrow mentioned there were plans to move stuff to a side panel.

**Comment by sean-mcmanus** [comment]
The CMake Tools extension has a cmake.statusbar setting. Is that sufficient?

**Reply by bobbrow** [reply]
> I remembered that @bobbrow mentioned there were plans to move stuff to a side panel.

Yes, it's this issue: https://github.com/microsoft/vscode-cmake-tools/issues/1582.  In the meantime you can use the cmake statusbar settings to reduce the clutter if you like.  Everything on the status bar should also be accessible via commands.

**Reply by H-G-Hristov** [reply]
> The CMake Tools extension has a cmake.statusbar setting. Is that sufficient?

Thank you. That's a start but not what I'd like. (manual settings via "advanced" option - isn't fun). I'd prefer a tighter integration between both extensions. Maybe some CMake settings could be moved into the same panel as CPP -> https://github.com/microsoft/vscode-cmake-tools/issues/1582

And maybe VSCode should have multi-row status bar :)

**Reply by sean-mcmanus** [reply]
Related VS Code issues are https://github.com/microsoft/vscode/issues/106032 and https://github.com/microsoft/vscode/issues/84258
